code/__HELPERS/mobs.dm

```diff
@@ -283,6 +291,10 @@ GLOBAL_LIST_EMPTY(species_list)
 				. = 0
 				break
 
+		if(target && (!(target in user.do_afters) || !(user in target.targeted_by)))

```

Comment:
No need to check for both. Just check for one of them.
If it happened that one of these checks-if-in-list were true and the other not then there would be a big problem. By default it shouldn't happen.